frontend/amundsen_application/static/js/pages/TableDetailPage/WriterDropdown/index.tsx

```diff
@@ -0,0 +1,166 @@
+
+export interface WriterDropdownProps {
+  tableApps: TableWriter[];
+}
+
+const isAirflowOrDatabricksApp = (appName) =>
+  appName.toLowerCase() === AIRFLOW || appName.toLowerCase() === DATABRICKS;
+
+const getSortedAppKinds = (apps: TableWriter[]) => {
+  const appKinds: string[] = [
+    ...new Set(apps.map((app) => (app.kind ? app.kind : PRODUCING))),
+  ];
+
+  const producingKind = appKinds.filter(
+    (kind) => kind.toLowerCase() === PRODUCING
+  );
+  const consumingKind = appKinds.filter(
+    (kind) => kind.toLowerCase() === CONSUMING
+  );
+
+  const sortedKinds: string[] = [];
+  sortedKinds.push(...producingKind, ...consumingKind);
+  sortedKinds.push(
+    ...appKinds.filter(
+      (kind) =>
+        kind.toLowerCase() !== PRODUCING && kind.toLowerCase() !== CONSUMING
+    )
+  );
+  return sortedKinds;
+};
+
+const getMenuItem = (app: TableWriter, menuItemClassName) => {
+  if (app.name.toLowerCase() === AIRFLOW) {
+    const [dagId, ...task] = app.id.split('/');
+    const taskId = task.join('/');
+    return (
+      <MenuItem
+        className={menuItemClassName}
+        href={app.application_url}
+        onClick={logClick}
+        target="_blank"
+      >
+        <div className="writer-dropdown-menu-item-row airflow-app">
+          <span className="section-title">{DAG_LABEL}</span>
+          <span className="menu-item-content">{dagId || NONE_VALUE}</span>
+        </div>
+        <div className="writer-dropdown-menu-item-row airflow-app">
+          <span className="section-title">{TASK_LABEL}</span>
+          <span className="menu-item-content">{taskId || NONE_VALUE}</span>
+        </div>
+      </MenuItem>
+    );
+  }
+  return (
+    <MenuItem
+      className={menuItemClassName}
+      href={app.application_url}
+      onClick={logClick}
+      target="_blank"
+    >
+      <div className="writer-dropdown-menu-item-row">
+        <span className="menu-item-content">
+          {app.id ? app.id : NONE_VALUE}
+        </span>
+      </div>
+    </MenuItem>
+  );
+};
+
+const WriterDropdown: React.FC<WriterDropdownProps> = ({
+  tableApps,
+}: WriterDropdownProps) => {
+  const tableAppName = tableApps[0].name.toLowerCase();
+
+  let image = '';
+  switch (tableAppName) {
+    case AIRFLOW:
+      image = '/static/images/airflow.jpeg';
+      break;
+    case DATABRICKS:
+      image = '/static/images/icons/logo-databricks.png';
+      break;
+    default:
+      image = '/static/images/icons/Application.svg';
+  }
+
+  tableApps.sort(
+    (a, b) => a.name.localeCompare(b.name) || a.id.localeCompare(b.id)
+  );
+  const appNames: string[] = [...new Set(tableApps.map((app) => app.name))];
+
+  const menuItems: JSX.Element[] = [];
+  let appsDefinedCount = 0;
+  appNames.forEach((name, nameIdx) => {
+    const appKinds = getSortedAppKinds(
+      tableApps.filter((app) => app.name === name)
+    );
+    appKinds.forEach((kind, kindIdx) => {
+      const sectionTitle = isAirflowOrDatabricksApp(tableApps[0].name)
```

Comment:
I wonder if it would be better to have just different menuItem components for the different apps.
Even if there is some duplication on the JSX markup, we would probably avoid some of the logic I we are using on this PR.

Think a branching like:
```
<Dropdown.Menu className="writer-dropdown-menu">
        {
  if (isAirflow) {
    return <AirFlowMenuItem />;
  } else if (isDatabricks) {
    return <DataBricksMenuItem  />;
  } else {
    return <GenericMenuItem />;
  }}
 </Dropdown.Menu>
```

Reply:
This is a good suggestion, it will make it easier/cleaner too if anyone else in the future wants to add special styling for a separate application. I'll make this change in the remaining work on this PR